### PR TITLE
psu: hangs while sending the last packet

### PR DESCRIPTION
**Pull request by HBuczynski**
JIRA: RTOS-70

## Description

If the last packet's size is not equal to `BUF_SIZE`, the `hid_write` sometimes returns -1.
Looking at wireshark logs, the device returns log `URB Status: No such file or directory (-ENOENT)`

I haven't found any information about packet's data alignment in imxrt106x Reference Manual. There is only note that there is allowed max 1024 bytes data per report.

I know that it is not the best way for solving this issue, but I haven't found the better one. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt106x).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: psu/psu.c [hunk 1]
```diff
@@ -159,7 +159,7 @@ static int sdp_writeFile(hid_device *dev, uint32_t addr, void *data, size_t size
 		/* print progress */
 		fprintf(stderr, "\r - Sent (%lu/%lu) %3.0f%% ", offset, size, ((float)offset / (float)size) * 100.0f);
 
-		if ((rc = hid_write(dev, b, n + 1)) < 0) {
+		if ((rc = hid_write(dev, b, BUF_SIZE)) < 0) {
```
Comment (nalajcie) [inline psu/psu.c]: please - zero-out remaining buffer (now we're sending trash from the previous packet)

**Review by nalajcie — CHANGES_REQUESTED** [review]
Unfortunately this hangs `psu` while writing kernel on `imx6ull`:

image

**Comment by nalajcie** [comment]
FYI, the `psu` hangs here:
image

File: psu/psu.c [hunk 2]
```diff
@@ -159,6 +160,12 @@ static int sdp_writeFile(hid_device *dev, uint32_t addr, void *data, size_t size
 		/* print progress */
 		fprintf(stderr, "\r - Sent (%lu/%lu) %3.0f%% ", offset, size, ((float)offset / (float)size) * 100.0f);
 
+		/* Report 2 size has to be align to 16, information define in HID Report Descriptor - ID 2 */
```
Comment (nalajcie) [inline psu/psu.c]: Really appreciate the comment regarding **why** this code is needed :). 
Btw, align -> aligned

**Review by nalajcie — DISMISSED** [review]
Tested on `imx6ull ` and it works. LGTM.

Editorial note: I would put "psu: fix hangup while..." as a commit message title for better readability (and automatic changelog generation in the future). No need to change it now, but keep it in mind in the future ;)

**Comment by HBuczynski** [comment]
I have analyzed the descriptors from HID device using wireshark.
It occurred that Report 2 which is responsible for sending data has `Report Size = 128` (bits) and `Report Count = 64`. 
So that the minimum Report Size should be multiplicity of 128 bits. The more information about HID protocol can be found https://usb.org/sites/default/files/hut1_2.pdf in chapter 4.6

The image below shows reports description:
image

Aligning size to 0x10 solves the problem both on `imxrt106x` and `imx6ull`.

**Review by nalajcie — APPROVED** [review]

**Pull request merged**